prov/sockets/src/sock_mr.c

```diff
@@ -118,14 +173,25 @@ struct sock_mr *sock_mr_verify_key(struct sock_domain *domain, uint64_t key,
 	}
 
 	fastlock_release(&domain->lock);
+
+	if (mr && domain->attr.mr_mode & FI_MR_RAW) {
+		if (mr->raw_key_len != raw_key_len) {
+			SOCK_LOG_ERROR("MR check failed - wrong raw key length\n");
+			mr = NULL;
+		} else if (0 != memcmp(mr->raw_key, raw_key, mr->raw_key_len)) {
```

Comment:
coding style uses forward logic.  I.e. we're comparing the results of memcmp to 0, not 0 against memcmp.